docs/adrs/007-minifront-v2-architecture.md

```diff
@@ -0,0 +1,261 @@
+
+### Architectural Evolution
+
+Minifront-v2 introduces several significant architectural improvements:
+
+#### 1. Feature-Sliced Design (FSD) Architecture
+
+```
+src/
+├── app/           # Application initialization & routing
+├── pages/         # Route-level page components
+├── widgets/       # Complex standalone UI widgets
+├── features/      # Business logic features
+├── entities/      # Business entities & domain logic
+└── shared/        # Shared utilities, services, stores
+```
+
+**Benefits:**
+
+- Clear separation of concerns by abstraction level
+- Improved scalability for larger feature sets
+- Better code organization and discoverability
+- Reduced coupling between components
+
+#### 2. MobX State Management
+
+**Divergence from original**: Replaces Zustand + TanStack Query with MobX stores
+
+```typescript
+// Original pattern (Zustand slice)
+export const useBalancesStore = create<BalancesSlice>((set, get) => ({
+  balances: [],
+  loading: false,
+  // ... manual state management
+}));
+
+// v2 pattern (MobX store)
+export class BalancesStore {
+  @observable balances: Balance[] = [];
+  @observable loading = false;
+
+  @computed get balancesByAccount() {
+    // Automatically memoized computed values
+  }
+}
+```
+
+**Benefits:**
+
+- **Performance**: Granular reactivity - only components observing changed data re-render
+- **Developer Experience**: Direct state mutations with automatic change detection
+- **Type Safety**: Full TypeScript inference without boilerplate
+- **Computed Values**: Memoized derivations that update only when dependencies change
+
+#### 3. Service Layer Architecture
+
+**New addition**: Centralized service layer for business logic separation
+
+```typescript
+// Services handle data operations
+export class PenumbraService {
+  async getBalances(): Promise<Balance[]> {
+    // gRPC communication logic
+  }
+}
+
+// Stores handle state management
+export class BalancesStore {
+  constructor(private service: PenumbraService) {}
+
+  async loadBalances() {
+    this.balances = await this.service.getBalances();
+  }
+}
+```
+
+**Benefits:**
+
+- Clean separation between data access and state management
+- Improved testability through dependency injection
+- Reusable business logic across stores
+- Easier mocking for testing
+
+#### 4. Dependency Injection Pattern
+
+**New addition**: Root store pattern with dependency injection
+
+```typescript
+export class RootStore {
+  readonly balancesStore: BalancesStore;
+  readonly transactionsStore: TransactionsStore;
+
+  constructor(private penumbraService: PenumbraService) {
+    this.balancesStore = new BalancesStore(this, penumbraService);
+    this.transactionsStore = new TransactionsStore(this, penumbraService);
+  }
+}
+```
+
+**Benefits:**
+
+- Centralized configuration and initialization
+- Cross-store communication through shared root
+- Easier testing with mock services
+- Clear application lifecycle management
```

Comment:
thought: "Centralized configuration and initialization" is more of a drawback and it contradicts to FSD. Having a root store proved to be a burden. I don't think stores need to communicate either, that's why they're "stores" – to simply store data. Any computation or modeling with this data can happen separately

suggestion: keep stores separate